src/credentials/providers/sharedCredentialsProvider.ts

```diff
@@ -73,17 +79,24 @@ export class SharedCredentialsProvider implements CredentialsProvider {
     public validate(): string | undefined {
         const expectedProperties: string[] = []
 
-        if (this.hasProfileProperty(SHARED_CREDENTIAL_PROPERTIES.ROLE_ARN)) {
+        if (
+            this.hasProfileProperty(SHARED_CREDENTIAL_PROPERTIES.ROLE_ARN) &&
+            this.hasProfileProperty(SHARED_CREDENTIAL_PROPERTIES.SOURCE_PROFILE)
```

Comment:
This change isn't tied to the telemetry changes. Please add it to the commit comments and PR description in case we need to look back at the commit history.